src/middleware.ts

```diff
@@ -0,0 +1,35 @@
+import { NextRequest, NextResponse } from 'next/server'
+import acceptLanguage from 'accept-language'
+import { fallbackLanguage, languages, cookieName } from './app/i18n/settings'
+
+acceptLanguage.languages(languages)
+
+export const config = {
+  // matcher: '/:language*'
+  matcher: ['/((?!api|_next/static|_next/image|assets|favicon.ico|sw.js).*)']
+}
+
+export function middleware(req: NextRequest) {
+  let language
+  if (req.cookies.has(cookieName)) language = acceptLanguage.get(req.cookies.get(cookieName)?.value)
+  if (!language) language = acceptLanguage.get(req.headers.get('Accept-Language'))
+  if (!language) language = fallbackLanguage
+
+  if (
+    !languages.some(loc => req.nextUrl.pathname.startsWith(`/${loc}`)) &&
```

Comment:
Extracting these individual conditionals out to a variable with a descriptive name would also help readability.